frontend/src/components/Gallery/Gallery.jsx

```diff
@@ -33,9 +33,13 @@ const Gallery = ({ images, apiType }) => {
 
     setFavImages(favImages.filter(favImg => favImg.id !== id));
   };
-
   const hasLike = (id) => favImages && favImages.find(favImg => favImg.id === id);
-
+  const seen = new Set();
+  images=images.filter((el) => {
```

Comment:
`images` is a react state here. You should never assign a value to the state directly. I will create a clean PR and mention you to show my solution. For now, you can either make this PR draft or close it.